lib/datasource/crate/index.ts

```diff
@@ -171,17 +171,48 @@ async function fetchRegistryInfo(
 
     const cacheKey = `crate-datasource/registry-clone-path/${registryUrl}`;
 
-    let clonePath: string = memCache.get(cacheKey);
-    if (!clonePath) {
+    // We need to ensure we don't run `git clone` in parallel. Therefore we store
+    // a promise of the running operation in the mem cache, which in the end resolves
+    // to the file path of the cloned repository.
+
+    const clonePathPromise: Promise<string> | null = memCache.get(cacheKey);
+    let clonePath: string;
+
+    if (clonePathPromise === null || clonePathPromise === undefined) {
```

Comment:
What would cause `clonePathPromise` to be `null` and in such a case would we want to retry? e.g. wouldn't it be either a `Promise` or `undefined`?

Reply:
This used to be `!clonePathPromise` (what I'd normally use to check for falsey values) written out because eslint doesn't allow that.

Flipped the if to get unified behavior for all falsy values and to remove the explicit checks.